guides/examples/Amazon SQS.md

```diff
@@ -1,3 +1,185 @@
+
+  1. Create a SQS queue (or use an existing one)
+  1. Configure our Elixir project to use Broadway
+  1. Define your pipeline configuration
+  1. Implement Broadway callbacks, so we can process incoming messages
+  1. Tuning the configuration (Optional)
+
+## Create a SQS queue
+
+Amazon provides a comprehensive [Step-by-step Guide](https://aws.amazon.com/getting-started/tutorials/send-messages-distributed-applications/)
+on creating SQS queues. In case you don't have an AWS account and want to
+test Broadway locally, use can easily install [ElasticMQ](https://github.com/softwaremill/elasticmq),
+which is a message queue system that offers a SQS-compatible query interface.
+
+## Configure the project
+
+In this guide we're going to use [BroadwaySQS](https://github.com/plataformatec/broadway_sqs),
+which is a Broadway SQS Connector provided by [Plataformatec](http://www.plataformatec.com).
+
+### Setting up dependencies
+
+Add `:broadway_sqs` to the list of dependencies in `mix.exs` along the HTTP
+client of your choice (defaults to `:hackney`):
+
+    def deps do
+      [
+        ...
+        {:broadway_sqs, "~> 0.1.0"},
+        {:hackney, "~> 1.9"},
+      ]
+    end
+
+## Define the pipeline configuration
+
+Like any other process-based behaviour, implementing the Broadway
+bahaviour is straightforward. The second argument of
+`Broadway.start_link/2` is the pipeline configuration. Assuming we
+want to consume messages from a queue called `my_queue`. The minimal
+configuration would be:
+
+    defmodule MyBroadway do
+      use Broadway
+      alias BroadwaySQS.{SQSProducer, ExAwsClient}
+
+      def start_link(_opts) do
+        Broadway.start_link(__MODULE__,
+          name: __MODULE__,
+          producers: [
+            default: [
+              module: SQSProducer,
+              arg: [
+                sqs_client: {ExAwsClient, [
+                  queue_name: "my_queue",
+                ]}
+              ]
+            ]
+          ],
+          processors: [default: []],
+          batchers: [default: []]
+        )
+      end
+
+      ...callbacks...
+    end
+
+The above configuration also assumes that you have the AWS credentials
+set up in your environment, for instance, by having the `AWS_ACCESS_KEY_ID`
+and `AWS_SECRET_ACCESS_KEY` environment variables set. If that's
+not the case, you will need to pass that information to the client so it
+can properly connect to the AWS servers. Here is how you can do it:
+
+    ...
+    sqs_client: {BroadwaySQS.ExAwsClient, [
+      queue_name: "my_queue",
+      config: [
+        access_key_id: "YOUR_AWS_ACCESS_KEY_ID",
+        secret_access_key: "YOUR_AWS_SECRET_ACCESS_KEY"
+      ]
+    ]}
+    ...
+
+
+For a full list of config options, please see [ExAws](https://hexdocs.pm/ex_aws/)
+documentation. You can also find adicional options for both `BroadwaySQS.SQSProducer`
+and `BroadwaySQS.ExAwsClient` in [BroadwaySQS](https://hexdocs.pm/broadway_sqs/) documentation.
```

Comment:
I am really unhappy with the amount of indirection. :( Let's have a call about this, I have some suggestions. :)

Reply:
@josevalim I agree. Just ping me on Slack when you're around.